src/cljs/rems/actions/request_comment.cljs

```diff
@@ -0,0 +1,134 @@
+(ns rems.actions.request-comment
+  (:require [re-frame.core :as rf]
+            [rems.actions.action :refer [action-form-view button-wrapper]]
+            [rems.atoms :refer [textarea]]
+            [rems.autocomplete :as autocomplete]
+            [rems.text :refer [text]]
+            [rems.util :refer [fetch post!]]))
+
+(rf/reg-fx
+ ::fetch-potential-commenters
+ (fn [user]
+   (fetch (str "/api/applications/reviewers") ; TODO separate API for commenters
+          {:handler #(do (rf/dispatch [::set-potential-commenters %])
+                         (rf/dispatch [::set-selected-commenters #{}]))
+           :headers {"x-rems-user-id" (:eppn user)}})))
+
+(rf/reg-event-fx
+ ::open-form
+ (fn [{:keys [db]} [_]]
+   (prn (get-in db [:identity :roles]))
+   (merge {:db (assoc db
+                      ::comment ""
+                      ::potential-commenters #{}
+                      ::selected-commenters #{})}
+          (when (contains? (get-in db [:identity :roles]) :approver) ; TODO handler role?
+            {::fetch-potential-commenters (get-in db [:identity :user])}))))
+
+
+
+;; TODO together with application.cljs
+(defn enrich-user [user]
+  (assoc user :display (str (:name user) " (" (:email user) ")")))
+
+(rf/reg-event-db
+ ::set-potential-commenters
+ (fn [db [_ users]]
+   (assoc db ::potential-commenters (map enrich-user users))))
+
+(rf/reg-sub ::potential-commenters (fn [db _] (::potential-commenters db)))
+
+(rf/reg-event-db
+ ::set-selected-commenters
+ (fn [db [_ users]]
+   (assoc db ::selected-commenters users)))
+
+(rf/reg-event-db
+ ::add-selected-commenter
+ (fn [db [_ user]]
+   (if (contains? (::selected-commenters db) user)
+     db
+     (update db ::selected-commenters conj user))))
+
+(rf/reg-event-db
+ ::remove-selected-commenter
+ (fn [db [_ user]]
+   (update db ::selected-commenters disj user)))
+
+(rf/reg-sub ::selected-commenters (fn [db _] (::selected-commenters db)))
+(rf/reg-sub ::comment (fn [db _] (::comment db)))
+
+(rf/reg-event-db
+ ::set-comment
+ (fn [db [_ value]] (assoc db ::comment value)))
+
+(defn- send-request-comment! [commenters user application-id comment description]
+  (post! "/api/applications/request_comment"
+         {:params {:application-id application-id
+                   :comment comment
+                   :recipients (map :userid commenters)}
+          :handler (fn [resp]
+                     ;; TODO use callbacks so no dependency?
+                     (rf/dispatch [:rems.application/set-status {:status :saved
```

Comment:
could trigger some sort of refresh-application event here, right?

Reply:
Do you mean a refresh-application which would replace the enter-application-page event it dispatches now?

Reply:
I think the use site in `application.cljs` should provide the refresh dispatching function to this component in a `on-success` callback. 

Set status can be refactored to each form. It does not need to stay in the `application.cljs`.